docs/en/latest/http3.md

```diff
@@ -0,0 +1,188 @@
+* providing zero round-trip time (0-RTT) when communicating with servers that clients already established connections to.
+
+APISIX currently supports HTTP/3 connections between downstream clients and APISIX. HTTP/3 connections with upstream services is not yet supported and contribution is welcomed.
+
+:::caution
+
+This feature is currently experimental and not recommended for production use.
+
+:::
+
+This document will show you how to configure APISIX to enable HTTP/3 connections between client and APISIX and document a few known issues.
+
+## Usage
+
+### Enable HTTP/3 in APISIX
+
+Enable HTTP/3 on port `9443` (or a different port) by adding the following configurations to APISIX's `config.yaml` configuration file:
+
+```yaml title="config.yaml"
+apisix:
+  ssl:
+    listen:
+      - port: 9443
+        enable_http3: true
+    ssl_protocols: TLSv1.3
+```
+
+:::info
+
+If you are deploying APISIX using Docker, make sure to allow UDP in the HTTP3 port, such as `-p 9443:9443/udp`.
+
+:::
+
+Then reload APISIX for configuration changes to take effect:
+
+```shell
+apisix reload
+```
+
+### Generate Certificates and Keys
+
+HTTP/3 requires TLS. You can leverage the purchased certificates or self-generate them, whichever applicable.
+
+To self-generate, first generate the certificate authority (CA) key and certificate:
+
+```shell
+openssl genrsa -out ca.key 2048 && \
+  openssl req -new -sha256 -key ca.key -out ca.csr -subj "/CN=ROOTCA" && \
+  openssl x509 -req -days 36500 -sha256 -extensions v3_ca -signkey ca.key -in ca.csr -out ca.crt
+```
+
+Next, generate the key and certificate with a common name for APISIX, and sign with the CA certificate:
+
+```shell
+openssl genrsa -out server.key 2048 && \
+  openssl req -new -sha256 -key server.key -out server.csr -subj "/CN=test.com" && \
+  openssl x509 -req -days 36500 -sha256 -extensions v3_req \
+  -CA ca.crt -CAkey ca.key -CAserial ca.srl -CAcreateserial \
+  -in server.csr -out server.crt
+```
+
+### Configure HTTPS
+
+Optionally load the content stored in `server.crt` and `server.key` into shell variables:
+
+```shell
+server_cert=$(cat server.crt)
+server_key=$(cat server.key)
+```
+
+Create an SSL certificate object to save the server certificate and its key:
+
+```shell
+curl -i "http://127.0.0.1:9180/apisix/admin/ssls" -X PUT -d '
+{
+  "id": "quickstart-tls-client-ssl",
+  "sni": "test.com",
+  "cert": "'"${server_cert}"'",
+  "key": "'"${server_key}"'"
+}'
+```
+
+### Create a Route
+
+Create a sample route to `httpbin.org`:
+
+```shell
+curl "http://127.0.0.1:9180/apisix/admin/routes" -X PUT -d '
+{
+  "id":"httpbin-route",
+  "uri":"/get",
+  "upstream": {
+    "type":"roundrobin",
+    "nodes": {
+      "httpbin.org:80": 1
+    }
+  }
+}'
+```
+
+### Verify HTTP/3 Connections
+
+Install [curl `>= 7.88.0`](https://curl.se/changes.html#7_88_0) for HTTP/3 support, or other curl that have HTTP/3 support.
```

Comment:
It is unknown why some of the latest versions have HTTP3 support while others don't.

![img_v3_02bb_5243c2cf-fbd7-4732-9079-94fc848a642g](https://github.com/apache/apisix/assets/39619599/6b889ff9-1f46-4e4f-9100-ee14f6dae7c8)

![img_v3_02bb_b4462261-fcf9-4881-a965-9ea39b0659dg](https://github.com/apache/apisix/assets/39619599/684a6f47-b988-4836-a23f-a364c8f060fe)

----

hence I'm considering changing this to:

```
Install [static-curl](https://github.com/stunnel/static-curl) or other curl that have HTTP/3 support.
```